libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -217,6 +219,17 @@ bool SyntaxChecker::visit(Throw const& _throwStatement)
 
 bool SyntaxChecker::visit(Literal const& _literal)
 {
+	if (_literal.token() == Token::UnicodeStringLiteral)
+	{
+		// Ensure it contains valid UTF-8 sequences only.
```

Comment:
I think we should do that for both string literals.

Reply:
So basically whoever wants to craft a non-UTF8 compliant string, evn if only via escapes, have to use `hex""`, not only in the case of strings, but in the case of bytes too?